lib/sycamore/sycamore/query/execution/sycamore_operator.py

```diff
@@ -110,20 +111,18 @@ def __init__(
         self.os_client_args = os_client_args
 
     def execute(self) -> Any:
-        assert self.logical_node.data and "index" in self.logical_node.data
-        result = self.context.read.opensearch(
-            os_client_args=self.os_client_args, index_name=self.logical_node.data["index"]
-        )
+        assert isinstance(self.logical_node, LoadData)
```

Comment:
can this ever be false

Reply:
same q for the other operators too

Reply:
Yes, because someone might have come along and manually done `node.logical_node = some_other_thing` after it was constructed. mypy isn't happy unless we make this assertion. I am not sure it would help to make the field immutable since I don't think (not sure) that mypy understands that.